### PR TITLE
Add local_return_period method to Impact objects and new tutorial

### PR DESCRIPTION
**Pull request by ValentinGebhart**
Changes proposed in this PR:
- New method `climada.engine.impact.Impact.local_return_period` that computes locally (i.e. per centroid) return periods for given threshold impact values (inputs to the method). Functionality analogous to `climada.hazard.base.Hazard.local_return_period`, but with impact values instead of intensity values.
- New tutorial `doc.tutorial.climada_util_local_exceedance_values.ipynb` that explains what is done in the methods `local_exceedance_frequency` and `local_return_period` of `Hazard` and `Impact` objects. First, a dummy hazard object is used to showcase the different settings of the methods. Then, the methods are applied to an exemplary `Hazard` and an exemplary `Impact` object.

### PR Author Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] Correct target branch selected (if unsure, select `develop`)
- [ ] Descriptive pull request title added
- [ ] Source branch up-to-date with target branch
- [ ] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [ ] [Tests][testing] updated
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]
- [ ] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/develop/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Review.html#reviewer-checklist) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/develop/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Testing.html
[linter]: https://climada-python.readthedocs.io/en/latest/guide/Guide_continuous_integration_GitHub_actions.html#static-code-analysis

### CODE REVIEW

**Review by spjuhel — COMMENTED** [review]
Very nice work! :+1: 

I have some minor comments on the tutorial, mostly typos/rephrasing, and maybe some clarification in the first part with the mock example hazard.

* "This is a tutorial to present and explain certain methods avaibale in `Hazard` and `Impact` objects. These methods" -> "This tutorial presents methods available in ... to compute ..."
* Cell "Compute local exceedance intensities": "We first sort the centroid's _intensit**i**es_ in descending order"
* -- : "100**<space>**years"
* -- : First centroid D then centroid C? Typo?
* -- : It took some time for me to understand. Maybe just briefly stating why we cumulate the frequencies, linking to exceedence.
* -- : last line "how **do** we want to estimate" (?)
* Cell "Option 1 (default setting)": "(here, 30 _years_)", same for "5 and 150" further in the sentence
* -- : I would swap the last two sentences on logarithmic scaling, to get the explanation right next to it (and remove the "however").
* Cell "option 2": the user wants to _guess_ -> _estimate_
* -- : "Also centroid with events intensity will be assigned zero exceedance intensity." Not sure what you mean here.
* Cell "option 4": "Here, return periods will be assigned the exceedance intensity of the next smaller of the observed return periods." -> I found this ambiguous. Is it the closest return period that is smaller or the smaller return period that is above? Maybe rephrase?
* Cell "Compute local return periods" : Rephrase. Maybe in the form "Using <method> you can find/plot the return period of events exceeding different intensities"
* Comparison of the described methods on a realistic ... : _real_ example rather than _realistic_ (being extra picky here ahah)
* "for 'test' return periods of" -> why 'test' ?
* "for a return period range from 5 to 250 years" -> either "for return periods ranging from" or "for a return period range of"
* "lie outside the range of _ob**s**erved_ values for this centroid (blue scatter points)"

**Comment by ValentinGebhart** [comment]
@spjuhel Thanks a lot for the comments, I have implemented them all. If you want, you can check what I added as a brief explanation why we sort the intensities und add up the frequencies in this specific way (second paragraph of Section Compute local exceedance intensities). I hope this makes things clearer :)

**Review by spjuhel — APPROVED** [review]
Nice! Happy to approve this PR :)

**Review by sarah-hlsn — APPROVED** [review]
This is a great tutorial! It really takes the user by the hand and goes through the different methods step-by-step. I particularly like the inclusion of a very simple mock hazard alongside real hazard objects. I have spotted some minor typo/grammar issues. Otherwise happy to approve if all checks are working. Here a couple of propsed changes:

- "Further below, we apply the methods to more real Hazard and Impact objects."  either _real --> realistic_ or _more real --> real_
- "The Hazard object consists of two events and has a spatial extend" _extend --> extent_
- "This resulting cumulative frequency for each intensity then yields the intensitiy's " _intensitiy's --> intensity's_
